### DISCUSSION TITLE
[RFC] Hey.. why aren't scraped tags/performers/studios automatically create-able on scrape?

### DISCUSSION THREAD

**Discussion by sleaze** [stashapp/stash#4787] [Q&A]
Stash is the best, except for this one thing..

The [official Stash docs](https://docs.stashapp.cc/beginner-guides/guide-to-scraping/) say:

> ! IMPORTANT !
> Tags are not created automatically. You need to click + icon near the tag name to create it for you locally and attach it to the scene. Once it’s created, it will match automatically in the future.

Why do we need to click the button?  I'd like the option to automatically trust upstream metadata providers to create tags / studios / performers on my behalf.  The clicking is super tedious and annoying, and I don't understand why I can't auto-scrape my entire collection from StashDB (to which I'm even a contributor).

I want the computer to work for me, not me be a button-clicking slave to it.

**Comment by sleetx** [comment]
Some people may not want certain tags in their library, and it would be annoying to have to delete those auto-created tags over and over again, each time metadata was scraped. 

With the current system, you have to OK the creation of the tag on your instance, and then it will be used going forward.

The former is a repeated inconvenience, and the latter is a one-time inconvenience.

**Comment by sleaze** [comment]
> Some people may not want certain tags in their library, and it would be annoying to have to delete those auto-created tags over and over again, each time metadata was scraped.
> 
> With the current system, you have to OK the creation of the tag on your instance, and then it will be used going forward.
> 
> The former is a repeated inconvenience, and the latter is a one-time inconvenience.

So make it a setting... I have a few hundred thousand items in the library, and can't be clicking for each one manually.  Right now, it's useless.  Given storage capacities today, I'm certainly not the only one in this predicament.

Can we make stash be useful for power users?

> The former is a repeated inconvenience, and the latter is a one-time inconvenience.

For. Every. New studio/performer/tag.  Over time, this adds up to a lot of manual toil for someone who isn't super sensitive due to the sheer scale.  Pretty please can I get some support in making this behavior configurable?

As of now I'm looking into a Puppeteer job to click the "+" button.  This feels wrong.

**Comment by echo6ix** [comment]
iirc tagger intentionally has more restraints for qc reasons. you're still going to get false positives via tagger.

but what you're describing sounds like it's already possible via the [Identify](https://docs.stashapp.cc/in-app-manual/tasks/identify/) task, no?

**Comment by sleaze** [comment]
`Identify` doesn't scrape, it only matches the path (or filename?) against existing tags, which is problematic and very low quality in many cases, e.g. "Walter White" gets tagged with "White".

**Comment by HereIam80** [comment]
> `Identify` doesn't scrape, it only matches the path (or filename?) against existing tags, which is problematic and very low quality in many cases, e.g. "Walter White" gets tagged with "White".

You are talking about Auto-Tag.
Identify is automated scraping from stashDB or some other sources as @echo6ix mentioned.
